apps/logging/request_logger.py

```diff
@@ -4,35 +4,68 @@
 from django.conf import settings
 from apps.dot_ext.loggers import get_session_auth_flow_trace
 
+CRITICAL = logging.CRITICAL
+FATAL = logging.FATAL
+ERROR = logging.ERROR
+WARNING = logging.WARNING
+WARN = logging.WARN
+INFO = logging.INFO
+DEBUG = logging.DEBUG
+NOTSET = logging.NOTSET
+
+
+def getLogger(name=None):
+    return RequestLogger(None, name) if name else BasicLogger()
```

Comment:
So when I originally created the `RequestLogger` and the `BasicLogger`. The idea was to always use the `RequestLogger` whenever possible, otherwise we'll miss out on the full log tracing. There are some rare cases where the `request` hasn't been established yet and we don't have access to it (like in some middleware) where we would have no choice but to use the `BasicLogger`. So with this line of code, you are basically always treating the `RequestLogger` as a `BasicLogger` since it is never getting passed the `request` object.

Reply:
the use pattern for bb2 logging is explicitly assign logger name to different loggers across bb2 areas, e.g. audit.x.y, and they actually are placed in the corresponding .py files, and the loggers can be turn off/on individually and their verbose levels can also be set individually, log events are partitioned per logger name, and logger tests also leverage logger channels to only capture a particular loger's events and validate them, carrying a logger reference with request object as in RequestLogger actually make the logger implied and all events fall into one category "audit_logging", this actually beat the logger use pattern, there might be scenarios that carrying a logger reference with the request object comes handy, but with python logger common practice, it might be a bit "anti-pattern", we could discuss this ...

Reply:
Yeah I remember we had this discussion before. Specifically around whether or not we would have any issues around switching the loggers to all use `audit.request_logger`. At the time I don't think we identified any issues. I agree that it might be a slight anti-pattern in Python. I don't think we actually really use the different loggers all that much. I know that we have them identified in the settings, but is there a good reason for making different categories of loggers use different log levels? Also there is still an avenue to use both names + request object. Kind of defeats the purpose of adding a preconfigured `logger` attached to the `request` object though.

Reply:
> Yeah I remember we had this discussion before. Specifically around whether or not we would have any issues around switching the loggers to all use `audit.request_logger`. At the time I don't think we identified any issues. I agree that it might be a slight anti-pattern in Python. I don't think we actually really use the different loggers all that much. I know that we have them identified in the settings, but is there a good reason for making different categories of loggers use different log levels? Also there is still an avenue to use both names + request object. Kind of defeats the purpose of adding a preconfigured `logger` attached to the `request` object though.

@nbragdon @JFU-GIT I'm thinking that using the same `logger_name` as `audit.request_logger` shouldn't cause any issues. There might be a dashboard search that would need modified, but there is also a `type` inside the message contents of the event. That is used in the dashboards.  Doing a quick look I didn't find any usage of the top level logger `name` being used in splunk.

This is compatible with the base.py settings log handlers (for `audit.*`). 

Also, I'm wondering if that `type` (message.type in the event) might have been a bad coding choice at the time and if the logger name of something like `audit.<type>` should have been used instead. Would be a pain related to searches/dashboards in splunk, if changing that now.

Reply:
@dtisza1 all logging goes to "audit.request_logger" will make it impossible to segregate log records by logger name, e.g., logger tests use that to capture only log records from a particular e.g. audit.authenticate.sls, otherwise, code need to added to scan records and pick the type of logging records and validate them etc.

note, leveraging logger name and use logger name to categorized log records is an important pattern from python logging, we might want to follow the pattern going forward...